src/crimson/osd/osd.cc

```diff
@@ -215,6 +217,33 @@ seastar::future<> OSD::_write_superblock()
   });
 }
 
+// this `to_string` sits in the `crimson::osd` namespace, so we don't brake
+// the language rule on not overloading in `std::`.
+static std::string to_string(const seastar::temporary_buffer<char>& temp_buf)
+{
+  return {temp_buf.get(), temp_buf.size()};
+}
+
+seastar::future<> OSD::_write_key_meta()
+{
+
+  if (auto key = local_conf().get_val<std::string>("key"); !std::empty(key)) {
+    return store.write_meta("osd_key", key);
+  } else if (auto keyfile = local_conf().get_val<std::string>("keyfile");
+             !std::empty(keyfile)) {
+    return read_file(keyfile).then([this] (const auto& temp_buf) {
+      // it's on a truly cold path, so don't worry about memcpy.
+      return store.write_meta("osd_key", to_string(temp_buf));
+    }).handle_exception([keyfile] (auto ep) {
+      logger().error("_write_key_meta: failed to handle keyfile {}: {}",
```

Comment:
how about using `or_terminate()`?

Reply:
The advantage of the current approach is the error msg (with the `keyfile` being explicitly printed).
On the other hand `or_terminate()` is would  be far more concise.

In summary: I'm **slightly** leaning towards `handle_exception()`; no strong feelings for sure.